test/unit/io_managers_test.py

```diff
@@ -268,46 +99,174 @@ def test_migrations_match_metadata(tmp_path, monkeypatch):
     assert True
 
 
-def test_error_when_handling_view_without_metadata(fake_pudl_sqlite_io_manager_fixture):
-    """Make sure an error is thrown when a user creates a view without metadata."""
-    asset_key = "track_view"
-    sql_stmt = "CREATE VIEW track_view AS SELECT * FROM track;"
-    output_context = build_output_context(asset_key=AssetKey(asset_key))
-    with pytest.raises(ValueError):
-        fake_pudl_sqlite_io_manager_fixture.handle_output(output_context, sql_stmt)
-
-
-@pytest.mark.skip(reason="SQLAlchemy is not finding the view. Debug or remove.")
-def test_handling_view_with_metadata(fake_pudl_sqlite_io_manager_fixture):
-    """Make sure an users can create and load views when it has metadata."""
-    # Create some sample data
-    asset_key = "artist"
-    artist = pd.DataFrame({"artistid": [1], "artistname": ["Co-op Mop"]})
-    output_context = build_output_context(asset_key=AssetKey(asset_key))
-    fake_pudl_sqlite_io_manager_fixture.handle_output(output_context, artist)
-
-    # create the view
-    asset_key = "artist_view"
-    sql_stmt = "CREATE VIEW artist_view AS SELECT * FROM artist;"
-    output_context = build_output_context(asset_key=AssetKey(asset_key))
-    fake_pudl_sqlite_io_manager_fixture.handle_output(output_context, sql_stmt)
-
-    # read the view data as a dataframe
-    input_context = build_input_context(asset_key=AssetKey(asset_key))
-    # print(input_context)
-    # This is failing, not sure why
-    # sqlalchemy.exc.InvalidRequestError: Could not reflect: requested table(s) not available in
-    # Engine(sqlite:////private/var/folders/pg/zrqnq8l113q57bndc5__h2640000gn/
-    # # T/pytest-of-nelsonauner/pytest-38/test_handling_view_with_metada0/pudl.sqlite): (artist_view)
-    fake_pudl_sqlite_io_manager_fixture.load_input(input_context)
-
-
-def test_error_when_reading_view_without_metadata(fake_pudl_sqlite_io_manager_fixture):
-    """Make sure and error is thrown when a user loads a view without metadata."""
-    asset_key = "track_view"
-    input_context = build_input_context(asset_key=AssetKey(asset_key))
-    with pytest.raises(ValueError):
-        fake_pudl_sqlite_io_manager_fixture.load_input(input_context)
+# TODO(rousik): is there difference between fake_sqlite_io_manager
+# and real sqlite_io_manager in terms of functionality?!!
+class PudlSQLiteIOManagerTest(unittest.TestCase):
```

Comment:
Why are you using unittest instead of pytest here?

Reply:
See other threads on the PR. I thought that `unitest` offers generally cleaner tests with less verbosity. However, if this breaks the convention (most of the code does use `pytest` even though there are some `unitest` based tests too), I can revert these changes.

Reply:
I do like the ability to put all of the assumptions/expectations about single class-under-test into a parallel unittest test-class, as this provides nice encapsulation and visual separation compared to the flat undifferentiated pytest style approach, but if we think consistency is more valuable (pytest vs unittest) I can revert these last changes.